authorizer-app-backend/src/main/java/org/radarbase/authorizer/doa/RestSourceUserRepositoryImpl.kt

```diff
@@ -23,61 +23,62 @@ import org.radarbase.authorizer.api.Page
 import org.radarbase.authorizer.api.RestOauth2AccessToken
 import org.radarbase.authorizer.api.RestSourceUserDTO
 import org.radarbase.authorizer.doa.entity.RestSourceUser
-import org.radarbase.jersey.exception.HttpBadGatewayException
 import org.radarbase.jersey.exception.HttpConflictException
 import org.radarbase.jersey.exception.HttpNotFoundException
 import org.radarbase.jersey.hibernate.HibernateRepository
 import java.time.Duration
 import java.time.Instant
 import java.time.temporal.ChronoUnit
+import java.util.*
 import javax.persistence.EntityManager
 
 class RestSourceUserRepositoryImpl(
     @Context em: Provider<EntityManager>,
 ) : RestSourceUserRepository, HibernateRepository(em) {
 
-    override fun create(token: RestOauth2AccessToken, sourceType: String): RestSourceUser = transact {
-        val externalUserId = token.externalUserId
-            ?: throw HttpBadGatewayException("Could not get externalId from token")
-
-        val queryString =
-            "SELECT u FROM RestSourceUser u where u.sourceType = :sourceType AND u.externalUserId = :externalUserId"
-        val existingUser = createQuery(queryString, RestSourceUser::class.java)
-            .setParameter("sourceType", sourceType)
-            .setParameter("externalUserId", externalUserId)
+    override fun create(user: RestSourceUserDTO): RestSourceUser = transact {
+        val existingUser = createQuery(
+            """
+            SELECT u
+            FROM RestSourceUser u
+            WHERE u.sourceType = :sourceType
+                AND u.userId = :userId
```

Comment:
Isn't the idea to look for existing users with the same fitbit account (externalId) ? or did we rename the externalUserId to userId?

Reply:
In this altered flow, the external ID is not yet known when we create the user. It is set after a new rest source user (combination MP user + source type) is made.